### PR TITLE
add method to reset the histogram

### PR DESCRIPTION
**Pull request by GlenTiki**
This is so I can work on something for https://github.com/mcollina/autocannon/issues/92

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -136,3 +136,25 @@ test('support >2e9', (t) => {
   t.ok(compare(instance.max(), recordValue), 'max match')
   t.end()
 })
+
+test('reset histogram', (t) => {
+  const instance = new Histogram(1, 100)
+  t.equal(instance.min(), 9223372036854776000, 'min is setup')
```
Comment (GlenTiki) [inline test.js:142]: @mcollina I discovered this tidbit when writing my tests, I have recreated it several times. should this be acceptable? its an empty instance.

File: package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
   "scripts": {
     "prepublish": "npm ls",
     "install": "node-pre-gyp install --fallback-to-build",
-   "test": "node-pre-gyp install --fallback-to-build && standard && tap test.js"
+    "test": "node-pre-gyp install --build-from-source && standard && tap test.js"
```
Comment (GlenTiki) [inline package.json]: Changed this command to ensure when running tests it always rebuilds. Previously it used any existing build, which wasn't optimal in dev.

File: hdr_histogram_wrap.cc [hunk 3]
```diff
@@ -186,3 +187,9 @@ NAN_METHOD(HdrHistogramWrap::Percentiles) {
 
   info.GetReturnValue().Set(result);
 }
+
+NAN_METHOD(HdrHistogramWrap::Reset) {
+  HdrHistogramWrap* obj = Nan::ObjectWrap::Unwrap<HdrHistogramWrap>(info.This());
+  hdr_reset(obj->histogram);
+  info.GetReturnValue().Set(true);
```
Comment (GlenTiki) [inline hdr_histogram_wrap.cc]: @mcollina Wasn't sure if I should or should not return here.

File: test.js (hunk 1 above)
Comment (mcollina) [reply test.js:142]: Not really.
However it is a bug in the C library. We might open a separate bug there.

File: hdr_histogram_wrap.cc (hunk 3 above)
Comment (mcollina) [reply hdr_histogram_wrap.cc]: IMHO we should return the histogram, or undefined.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: test.js (hunk 1 above)
Comment (GlenTiki) [reply test.js:142]: 👍

File: hdr_histogram_wrap.cc (hunk 3 above)
Comment (GlenTiki) [reply hdr_histogram_wrap.cc]: I'll return the histogram.

**Review by mcollina — APPROVED** [review]
LGTM

File: test.js (hunk 1 above)
Comment (GlenTiki) [reply test.js:142]: I've done some research on this, I don't believe it the issue we thought it. The min value should be initialised to positive infinity, and max value to negative infinity. It means the check for greater than or less than values are not complex logic trees, and you just use the single variables from initialisation in implementation.

Comment (GlenTiki) [reply test.js:142]: & relevant issue: https://github.com/HdrHistogram/HdrHistogram/issues/54

**Pull request merged**